Model/Service/QuoteAddressService.php

```diff
@@ -0,0 +1,253 @@
+     * @return void
+     */
+    protected function maybeFillShippingAddressFields($quote)
+    {
+        $address = $quote->getShippingAddress();
+        if ($address->getId() === null) {
+            $address->setFirstname('unknown');
+            $address->setLastname('unknown');
+            $address->setTelephone('unknown');
+            $address->setEmail('no-reply@example.com');
+            $address->setStreet('unknown');
+            $quote->setShippingAddress($address);
+        }
+    }
+
+    /**
+     * If we didn't find any default billing address we fill the empty fields
+     * required for quote validation
+     *
+     * @param ShippingAddressRequestInterface $shipping_address
+     * @param Quote $quote
+     *
+     * @return void
+     */
+    protected function maybeFillBillingAddressFields(ShippingAddressRequestInterface $shipping_address, $quote)
+    {
+        $address = $quote->getBillingAddress();
+        if ($address->getId() === null) {
+            $address->setFirstname('unknown');
+            $address->setLastname('unknown');
+            $address->setTelephone('unknown');
+            $address->setEmail('no-reply@example.com');
+            $address->setStreet('unknown');
+            $address->setCountryId($shipping_address->getCountryCode());
+            $address->setPostcode($shipping_address->getPostalCode());
+            $address->setCity($shipping_address->getCity());
+            $address->setRegion($shipping_address->getState());
+            $quote->setBillingAddress($address);
+        }
+    }
+
+    public function assignAddressToQuote($shippingAddress, $cart)
+    {
+        try {
+            $this->shippingAddressManagement->assign($cart->getId(), $shippingAddress);
+        } catch (\Exception $e) {
+            $this->logger->addDebug(__METHOD__ . '|9.1|' .  $e->getMessage());
+            throw new ExpressMethodsException(__(
+                'Assign Shipping Address to Quote failed.'
+            ));
+        }
+        $this->quoteRepository->save($cart);
+
+        return $cart;
+    }
+
+    /**
+     * Set Shipping Address on SaveOrder
+     */
+    public function setShippingAddress(&$quote, $data)
+    {
+        $this->logger->addDebug(__METHOD__ . '|1|');
+
+        $shippingAddress = $this->processAddressFromWallet($data);
+        $quote->getShippingAddress()->addData($shippingAddress);
+        $quote->setShippingAddress($quote->getShippingAddress());
+
+        $errors = $quote->getShippingAddress()->validate();
+        return $this->setCommonAddressProceed($errors, 'shipping');
+    }
+
+    /**
+     * Set Billing Address on SaveOrder
+     */
+    public function setBillingAddress(&$quote, $data)
+    {
+        $this->logger->addDebug(__METHOD__ . '|1|');
+
+        $billingAddress = $this->processAddressFromWallet($data, 'billing');
+        $quote->getBillingAddress()->addData($billingAddress);
+        $quote->setBillingAddress($quote->getBillingAddress());
```

Comment:
what's this for?

Reply:
setShippingAddress and setBillingAddress are used in Applepay/SaveOrder
![image](https://user-images.githubusercontent.com/106905746/227144627-ae297f41-e55c-401b-91dd-beeb7dcef509.png)
I can not used too "addAddressToQuote" because this function set just setCountryId, setPostcode, setCity and setRegion rest of the fields being completed with default data ('unknown')

Reply:
Yes, you are right